pennylane/transforms/adjoint.py

```diff
@@ -102,9 +102,10 @@ def wrapper(*args, **kwargs):
             try:
                 op.adjoint()
             except NotImplementedError:
-                # Decompose the operation and adjoint the result.
+                # Expand the operation and adjoint the result.
                 # We do not do anything with the output since
-                # decomposition will automatically queue the new operations.
-                adjoint(op.decomposition)(wires=op.wires)
+                # calling adjoint on the expansion will automatically
+                # queue the new operations.
+                adjoint(op.expand)()

```

Comment:
Modify `qml.adjoint()` to use `expand` rather than `decomposition`.

Reply:
This will behave _slightly_ differently than how it behaves currently. `op.expand` will append a new `tape` to the current context, whereas `op.decompose` will directly append the operations. I originally picked decompose for this reason.

Reply:
Yep, I was worried when making this change. However, it appears to work out-of-the-box, due to the nesting ability of the tape.

Given

```
op1 -- op2.inv -- op3
```

using `expand` gives


```
op1 -- ||op2a--op2b||  -- op3
```

(where || indicates a nested tape) whereas using `decomposition` inserts it into the same level:


```
op1 -- op2a -- op2b  -- op3
```

So works out of the box, with _perhaps_ a minimal speed penalty?

Reply:
Yeah, actually the more I think about it, the more I prefer having it queue a tape instead.

If you had

```
---op1----custom_op----op3-----
```

Previously, adjoint would give

```
---op1-----op_a---op_b---op_c----op3----
```

But now gives
```
---op1--||---op_a---op_b---op_c---||-op3----
```

So now, replacing a single op equates to adding a single tape, instead of multiple ops. That feels nicer in my math brain, so I like it.

Can we also update the docstring to account for this change?

Reply:
> Can we also update the docstring to account for this change?

Will do!